cpp/src/parquet/file_reader.cc

```diff
@@ -119,15 +133,57 @@ class SerializedRowGroup : public RowGroupReader::Contents {
 
     std::shared_ptr<ArrowInputStream> stream =
         properties_.GetStream(source_, col_start, col_length);
+
+#ifdef PARQUET_ENCRYPTION
```

Comment:
Can we remove these `ifdef` and use conditionals based on the presence of crypto variables? In this case we can check based off `col->crypto_metadata()`?

Reply:
Yep, looks like col->crypto_metadata() check does the job here. We'll also handle all other similar ifdefs.

Reply:
I have removed some but not completely. I'm not sure if it matches your expectation. Can you please take a look again?

Reply:
missed this question, please see the comment below.